src/multibody/parser/urdf-with-geometry.hxx

```diff
@@ -101,31 +100,38 @@ namespace se3
     inline void parseTreeForGeom(::urdf::LinkConstPtr link,
                                  const Model & model,
                                  GeometryModel & geom_model,
-                                 const std::string & meshRootDir,
+                                 const std::vector<std::string> & package_dirs,
                                  const bool rootJointAdded) throw (std::invalid_argument)
     {
 
       // start with first link that is not empty
-      if(link->collision)
```

Comment:
You should first check if the link has a collision information or not. Furthermore, if the link has no inertial information but some collision objects, we must keep this information.